crates/turborepo-lib/src/commands/link.rs

```diff
@@ -308,19 +364,25 @@ pub async fn link(
     }
 }
 
-fn should_enable_caching() -> Result<bool> {
+fn should_enable_caching() -> Result<bool, Error> {
     let theme = ColorfulTheme::default();
-    Ok(Confirm::with_theme(&theme)
+
+    Confirm::with_theme(&theme)
         .with_prompt(
             "Remote Caching was previously disabled for this team. Would you like to enable it \
              now?",
         )
         .default(true)
-        .interact()?)
+        .interact()
+        .map_err(|_| Error::UserCanceled)
```

Comment:
Can we keep the underlying `io::Error` as part of this error? [Some Windows terminal emulators](https://code.google.com/archive/p/mintty/issues/56) do not play nicely interactive CLIs and I'd like to provide that error instead of a generic "user canceled" error.